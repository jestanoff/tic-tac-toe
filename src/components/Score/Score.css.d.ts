// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'active': string;
  'bluePrimary': string;
  'container': string;
  'fontPrimary': string;
  'greyPrimary': string;
  'greyTertiary': string;
  'mark': string;
  'score': string;
}
export const cssExports: CssExports;
export default cssExports;
