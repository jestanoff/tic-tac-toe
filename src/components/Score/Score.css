@value bluePrimary, greyPrimary, greyTertiary from "css/colors.css";
@value fontPrimary from "css/font.css";

.mark {
  display: inline-block;
  height: 28px;
  width: 28px;
}

.container {
  background: white;
  border-bottom: 3px solid transparent;
  box-shadow: 0 0 1px greyTertiary;
  box-sizing: border-box;
  display: inline-block;
  margin: 1%;
  padding: 5px 4% 0 2%;
  width: 48%;
}

.active {
  composes: container;
  border-bottom-color: bluePrimary;
  box-shadow: 0 0 6px greyTertiary;
}

.score {
  color: greyPrimary;
  float: right;
  font-family: fontPrimary;
  font-size: 20px;
  padding-top: 2px;
}
