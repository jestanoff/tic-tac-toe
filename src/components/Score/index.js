import React, { memo } from 'react';
import { bool, number, oneOfType, string } from 'prop-types';
import Mark from 'components/Mark';
import { DARK_GRAY } from 'constants';
import styles from './Score.css';

const Score = ({ isActive, score, symbol }) => (
  <div className={isActive ? styles.active : styles.container}>
    <span className={styles.mark}>
      <Mark type={symbol} overrideColor={DARK_GRAY} />
    </span>
    <span className={styles.score}>{score}</span>
  </div>
);

Score.propTypes = {
  isActive: bool.isRequired,
  score: oneOfType([number, string]).isRequired,
  symbol: string.isRequired,
};

export default memo(Score);
