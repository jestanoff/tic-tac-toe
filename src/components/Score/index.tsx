import * as React from 'react';
import Mark from 'components/Mark';
import { DARK_GRAY } from '../../constants';
import styles from './Score.css';

type ScoreProps = {
  isActive: boolean,
  score: number | string,
  symbol: string,
};

const Score = ({ isActive, score, symbol }: ScoreProps) => (
  <div className={isActive ? styles.active : styles.container} data-testid={`score-for-player-${symbol}`}>
    <span className={styles.mark}>
      <Mark type={symbol} overrideColor={DARK_GRAY} />
    </span>
    <span aria-live='polite' className={styles.score} title={`Player ${symbol} has ${Number(score) || 0} points`}>{score}</span>
  </div>
);

export default React.memo(Score);
