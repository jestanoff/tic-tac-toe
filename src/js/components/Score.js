import React from 'react';
import PropTypes from 'prop-types';
import Mark from './Mark';
import styles from '../../css/score.css';
import { DARK_GRAY } from '../constants/constants';

const Score = ({ isActive, score, symbol }) => (
    <div className={ isActive ? styles.active : styles.container }>
        <span className={ styles.mark } ><Mark type={ symbol } overrideColor={ DARK_GRAY } /></span>
        <span className={ styles.score } >{ score }</span>
    </div>
);

Score.propTypes = {
    isActive: PropTypes.bool.isRequired,
    score: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
};

export default Score;
