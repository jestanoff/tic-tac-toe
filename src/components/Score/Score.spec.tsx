import * as React from 'react';
import { render, screen } from '@testing-library/react';
import Score from './index';

describe('<Score />', () => {
  test('should render active score for cross', () => {
    const { asFragment } = render(<Score isActive score="?" symbol="cross" />);

    expect(screen.getByTestId('score-for-player-cross')).toHaveClass('active');
    expect(screen.queryByTestId('score-for-player-cross')).not.toHaveClass('container');
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render not active score', () => {
    render(<Score isActive={false} score="-" symbol="cross" />);

    expect(screen.getByTestId('score-for-player-cross')).toHaveClass('container');
    expect(screen.queryByTestId('score-for-player-cross')).not.toHaveClass('active');
  });

  test('should render 5 score points', () => {
    render(<Score isActive score={5} symbol="circle" />);

    expect(screen.getByTitle('Player circle has 5 points')).toHaveTextContent('5');
  });
});
