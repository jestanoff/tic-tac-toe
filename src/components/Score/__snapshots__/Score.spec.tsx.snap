// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> should render active score for cross 1`] = `
<DocumentFragment>
  <div
    class="active"
    data-testid="score-for-player-cross"
  >
    <span
      class="mark"
    >
      <div
        class="container"
      >
        <svg
          class="svg"
          data-testid="cross-mark"
          preserveAspectRatio="xMinYMin meet"
          version="1.1"
          viewBox="0 0 500 500"
        >
          <line
            stroke="#444444"
            stroke-width="50"
            x1="130"
            x2="130"
            y1="130"
            y2="130"
          >
            <animate
              attributeName="x2"
              attributeType="XML"
              dur="1ms"
              fill="freeze"
              from="130"
              id="first"
              repeatCount="1"
              to="370"
            />
            <animate
              attributeName="y2"
              attributeType="XML"
              dur="1ms"
              fill="freeze"
              from="130"
              repeatCount="1"
              to="370"
            />
          </line>
          <line
            opacity="0"
            stroke="#444444"
            stroke-width="50"
            x1="370"
            x2="370"
            y1="130"
            y2="130"
          >
            <animate
              attributeName="opacity"
              attributeType="XML"
              begin="0s"
              dur="1ms"
              fill="freeze"
              from="0"
              repeatCount="1"
              to="1"
            />
            <animate
              attributeName="x2"
              attributeType="XML"
              begin="0s"
              dur="1ms"
              fill="freeze"
              from="370"
              repeatCount="1"
              to="130"
            />
            <animate
              attributeName="y2"
              attributeType="XML"
              begin="0s"
              dur="1ms"
              fill="freeze"
              from="130"
              repeatCount="1"
              to="370"
            />
          </line>
        </svg>
      </div>
    </span>
    <span
      aria-live="polite"
      class="score"
      title="Player cross has 0 points"
    >
      ?
    </span>
  </div>
</DocumentFragment>
`;
